cylc/flow/data_store_mgr.py

```diff
@@ -1335,6 +1349,11 @@ def update_workflow(self):
             w_delta.is_queued_total = is_queued_total
             delta_set = True
 
+            for state, tp_queue in self.latest_state_tasks.items():
+                if tp_queue:
```

Comment:
I think this is never `None` as we initialize each `state: deque(...)` in the `__init__` function?

Reply:
`tp_queue` is empty it doesn't set it .. otherwise you'd get `failed: []` in the feed.

Which would you prefer on the front end processing? list and item always there? or not?

Reply:
Ah, true, it also checks for empty. Sorry @dwsutherland , confused it with JavaScript :smile: easier the way you did IMO :+1:

Reply:
All good either way, up to you 😃